frame/transaction-payment/rpc/runtime-api/src/lib.rs

```diff
@@ -30,5 +32,37 @@ sp_api::decl_runtime_apis! {
 	{
 		fn query_info(uxt: Block::Extrinsic, len: u32) -> RuntimeDispatchInfo<Balance>;
 		fn query_fee_details(uxt: Block::Extrinsic, len: u32) -> FeeDetails<Balance>;
+		fn query_call_info(encoded_call: Bytes) -> Result<RuntimeDispatchInfo<Balance>, DispatchError>;
+		fn query_weight_to_fee(encoded_call: Bytes) -> Result<Balance, DispatchError>;
+		// fn query_call_info(call: GetDispatchInfo, len: u32) -> RuntimeDispatchInfo<Balance>;
+		// fn query_weight_to_fee(call: GetDispatchInfo) -> Balance;
```

Comment:
Possible option, but wont work with `api.rpc.state.call` for light clients

Reply:
It doesn't work because this isn't legal Rust. `GetDispatchInfo` is a trait. You need to use a concrete type here. Just use the chains overarching `Call` type and you will be fine (you will need to add it as a type parameter). And of course it works with `state_call`. The `Call` type is encoded in the runtime's metadata and a client can encode any data it passes to `state_call` according to it.